### PR TITLE
Fix compile on ghc 8.8.

### PR DESCRIPTION
**Pull request by mikesperber**
These are all MonadFail-related.

### CODE REVIEW

**Comment by blackgnezdo** [comment]
Thanks for the contribution! Could you fix the error reported by the build?

```
tensorflow-ops             > /tfhs/tensorflow-ops/src/TensorFlow/Variable.hs:96:23: error: [-Wname-shadowing, -Werror=name-shadowing]
tensorflow-ops             >     This binding for ‘h’ shadows the existing binding
tensorflow-ops             >       bound at src/TensorFlow/Variable.hs:94:10
tensorflow-ops             >    |
tensorflow-ops             > 96 |             (Variable h Nothing :: Variable a) -> (h, a)
tensorflow-ops             >    |                       ^
```

File: tensorflow-ops/tests/EmbeddingOpsTest.hs [hunk 1]
```diff
@@ -161,7 +161,11 @@ instance Arbitrary a => Arbitrary (LookupExample a) where
         let maxDim = fromIntegral (ceiling doubleMaxDim :: Int64)
             doubleMaxDim :: Double
             doubleMaxDim = 100 ** (1 / fromIntegral rank)
-        shape@(firstDim : _) <- vectorOf rank (choose (1, maxDim))
+        shape <- vectorOf rank (choose (1, maxDim))
+        let firstDim =
+              case shape of
+                firstDim : _ -> firstDim
+                _ -> undefined -- avoid MonadFail
```
Comment (blackgnezdo) [inline tensorflow-ops/tests/EmbeddingOpsTest.hs:168]: I think you chose to use head below, maybe follow suite here?

File: tensorflow-ops/src/TensorFlow/Variable.hs [hunk 2]
```diff
@@ -90,7 +90,11 @@ initializedVariable' :: forall a m v . (MonadBuild m, TensorType a)
                     => OpParams -> Tensor v a -> m (Variable a)
 initializedVariable' params initializer = do
     -- The shape is not known initially.
-    (Variable h Nothing :: Variable a) <- variableInternal params Nothing
+    variables <- variableInternal params Nothing
+    let (h, a) =
+          case variables of
+            (Variable h Nothing :: Variable a) -> (h, a)
+            _ -> undefined -- avoid MonadFail
```
Comment (blackgnezdo) [inline tensorflow-ops/src/TensorFlow/Variable.hs:97]: Could you make this a tad more helpful when it fails?
`error "variableInternal is empty"` should be easier to decipher.

Comment (mikesperber) [reply tensorflow-ops/src/TensorFlow/Variable.hs]: @blackgnezdo Thanks for the comments - good points all. Revised the patch, hopefully better.

**Comment by blackgnezdo** [comment]
GHC is still unhappy, could you fix these too?
```
tensorflow-ops             > /tfhs/tensorflow-ops/tests/GradientTest.hs:563:9: error: [-Wname-shadowing, -Werror=name-shadowing]
tensorflow-ops             >     This binding for ‘l’ shadows the existing binding
tensorflow-ops             >       bound at tests/GradientTest.hs:560:9
tensorflow-ops             >     |
tensorflow-ops             > 563 |         l <- TF.gradients f'x [w] -- take gradient again (this time over w)
tensorflow-ops             >     |         ^
tensorflow-ops             > 
tensorflow-ops             > /tfhs/tensorflow-ops/tests/GradientTest.hs:653:9: error: [-Wname-shadowing, -Werror=name-shadowing]
tensorflow-ops             >     This binding for ‘l’ shadows the existing binding
tensorflow-ops             >       bound at tests/GradientTest.hs:650:9
tensorflow-ops             >     |
tensorflow-ops             > 653 |         l <- TF.gradients f'x [w] -- take gradient again (this time over w)
tensorflow-ops             >     |         ^
```

**Comment by mikesperber** [comment]
Given that monadic syntax is inherently sequential, this makes no sense to me. But fixed it.

**Pull request merged**